examples/simple_autopilot_private_connect_gateway/iam.tf

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_project_iam_member" "viewer" {
+  for_each = {
+    for index, user_perm in var.user_permissions :
```

Comment:
IIUC for each user in var.user_permissions, you want to grant "roles/gkehub.admin", "roles/gkehub.gatewayAdmin", "roles/container.viewer"? It might be better to get a set of all users first and then setproduct with the list of roles to foreach over one google_project_iam_member resource.